core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala

```diff
@@ -914,6 +920,10 @@ class GroupCoordinator(val brokerId: Int,
       if (isPending) {
         debug(s"Pending member $memberId has been removed after session timeout expiration.")
         group.removePendingMember(memberId)
+        if (group.is(PreparingRebalance)) {
+          debug(s"group was in preparing rebalance stage. will attempt to complete join now.")
```

Comment:
I'm not sure how useful it is to log the fact that we're checking completion, but if you disagree, we should at least add the groupId to the message. Similarly the log line above on 921 should include the groupId.